### DISCUSSION TITLE
VS Code debug test with v3

### DISCUSSION THREAD

**Discussion by careri** [xunit/xunit#3019] [Question] [answered]
Hi, I tried searching the web for a solution but have yet to find one.

Packages:
```xml
    <PackageReference Update="xunit.v3" Version="0.3.0-pre.18" />
    <PackageReference Update="xunit.runner.visualstudio" Version="3.0.0-pre.30" PrivateAssets="All" />
```

I migrated from v2 to v3 to get access to the culture property in xunit.runner.json.
I can run the tests from VS Code and they show up in the test explorer. But I can't debug.

As far as I can tell the process launched by VS Code is no longer called testhost.exe, but <project_name>.exe. So the debugger fails to find the correct process.
![image](https://github.com/user-attachments/assets/94aad4c6-a085-45fb-8ebf-e49465ef0512)

Any ideas?

**Comment by bradwilson** [comment]
Hmm, odd. It definitely still works with 0.2.0-pre.69 and 3.0.0-pre.24, but ~updating either one to the newer version breaks debugging~. **Edit:** Updating to 0.3.0-pre.18 is fine, it's the update to 3.0.0-pre.30 that is causing the problems.

I'll see if I can figure out what's going on.

**Reply by bradwilson** [reply]
Opened an issue: https://github.com/xunit/visualstudio.xunit/issues/415

**Reply by bradwilson** [reply, marked as answer]
The fix is now available with `3.0.0-pre.31` https://xunit.net/docs/using-ci-builds

**Reply by careri** [reply]
Perfect, thanks for the prompt fix :)
